### PR TITLE
Expose Configuration as individual fields

### PR DESCRIPTION
**Pull request by francoislg**

### CODE REVIEW

File: template.tpl [hunk 1]
```diff
@@ -443,25 +443,65 @@ ___TEMPLATE_PARAMETERS___
...
+        "name": "analyticsEndpoint",
+        "displayName": "Analytics Endpoint URL",
+        "macrosInSelect": true,
+        "selectItems": [
+          {
+            "value": "https://usageanalytics.coveo.com/",
+            "displayValue": "Coveo Cloud"
+          },
+          {
+            "value": "http://usageanalyticshipaa.cloud.coveo.com/",
```
Comment (dlafreniere) [inline template.tpl]: https

File: template.tpl [hunk 2]
```diff
@@ -781,13 +816,13 @@ const loadCoveoAnalyticsScript = () => {
...
   
-  const conf = data.configuration;
-  coveoua("init", conf.apiKey, conf.analyticsEndpoint);
+  coveoua("init", data.apiKey, data.analyticsEndpoint);
   coveoua("onLoad", function() {
     log('Coveo Analytics Initialized');
   });
   
-  const url = "https://static.cloud.coveo.com/coveo.analytics.js/" + conf.scriptVersion + "/coveoua.js";
+  const scriptVersion = data.scriptVersion || "1.0";
```
Comment (dlafreniere) [inline template.tpl]: can we detect a version that doesn't exist?

Comment (francoislg) [reply template.tpl]: Following your comment, I tried putting `2.0`.

I get a "403" in the console, and it stops the execution of the tag in `injectScript`. 🤔

I don't think there is an easy way to validate that the version exists, any idea? The 403 might be enough 🤔

Comment (dlafreniere) [reply template.tpl]: I would have expected a 404 😕

Comment (francoislg) [reply template.tpl]: It's the CDN that somehow returns this error:

https://static.cloud.coveo.com/coveo.analytics.js/2.0/coveoua.js

The same thing happens with the `coveo-search-ui` project 🤔 

https://static.cloud.coveo.com/searchui/invalid/js/CoveoJsSearch.Lazy.min.js

**Pull request merged**